FiveCalls/FiveCalls/Store.swift

```diff
@@ -58,7 +58,12 @@ class Store: ObservableObject {
         case let .SetContacts(contacts):
             state.contacts = contacts
         case let .SetDistrict(district):
+            let oldDistrict = state.district
             state.district = district
+
+            if oldDistrict != district {
+                dispatch(action: .FetchMessages)
```

Comment:
nice, this must be in the next run loop so you didn't need the dispatch queue hack?

Reply:
yep! much better this way